doc/copp/CoPP Config and Management.md

```diff
@@ -0,0 +1,211 @@
+# Introduction
+
+The scope of this document is to provide the requirements and a high-level design proposal for Sonic Control Plane Policing configuration and management.
+
+# Requirements
+
+The following are the high level requirements for CoPP/TRAP managment
+
+1. CoPP Tables shall be part of the Config DB. Currently this is only present in APP DB
+2. User shall be able to configure, modify CoPP table entries for various policing requirements
+3. System shall be backward compatible with old Config DB without any CoPP Tables
+4. A VS test must be added to verify the Config DB CoPP tables
+
+# Design Proposal
+
+## Current behaviour
+
+During SWSS start, the prebuilt [copp.json](https://github.com/Azure/sonic-swss/blob/201911/swssconfig/sample/00-copp.config.json) file is loaded as part of start script [swssconfig.sh](https://github.com/Azure/sonic-buildimage/blob/201911/dockers/docker-orchagent/swssconfig.sh) and written to APP DB. [CoppOrch](https://github.com/Azure/sonic-swss/blob/201911/orchagent/copporch.cpp) then translates it to Host trap/policer tables and programmed to SAI.
+
+## Proposed behaviour
+
+With the new proposal, the CoPP tables shall be loaded to Config DB instead of APP DB. The default CoPP json file shall be prebuilt to the image and loaded during initialization. Any Config DB entries present shall be configured overwriting the default CoPP tables. This also ensures backward compatibility. At a high-level, the following are the expected changes:
+
+### Schema Changes
+
+A new schema is defined for COPP tables that seperates Queue/Policer groups and Traps. More details are in the "Examples" section
+
+### Config DB
+```
+key = "COPP_GROUP|name"
+queue         = number; strict queue priority. Higher number means higher priority.
```

Comment:
Does it make sense to move the queue to COPP_TRAP table?  Then COPP_GROUP reference could become a POLICER_TABLE reference in COPP_TRAP.

Decoupling queue from policer helps avoid extra COPP_GROUPs from being defined when different policer rates are required but same queue is needed or visa versa.

Reply:
This could require further changes and also may affect warmboot. If not with major advantage, prefer to keep it this way